### ISSUE TITLE
Should hydrostatics be computed with full mesh or clipped mesh?

### ISSUE THREAD

**Issue by mancellin** [capytaine/capytaine#197]
> Edit: changed the code sample to define the center of mass of body before calling add_all_rigid_body_dofs(). In this way, the rotation dofs are defined around the center of mass and not the default (0, 0, 0).

So when I compute the inertia matrix before  calling `keep_immersed_part()`?Otherwise I can't get the inertia matrix for the whole rigid body.

_Originally posted by @ykshif in https://github.com/capytaine/capytaine/issues/188#issuecomment-1231794985_

**Comment by mancellin** [comment]
Most hydrostatics magnitudes are computed on the clipped mesh (the `immersed_part()`).
For the inertia matrix, it would mean taking only the immersed mass into account, which is usually not what we want.
In `FloatingBody.compute_hydrostatics()`, the inertia matrix is computed on the clipped mesh, but I'm not sure it makes sense.

I guess, we should recommend to run the commands in the following order:
```python
body = cpt.Sphere()
body.center_of_mass = (0, 0, -0.1)
body.add_all_rigid_body_dofs()
body.inertia_matrix = body.compute_rigid_body_inertia()
body.keep_immersed_part()
body.hydrostatic_stiffness = body.compute_hydrostatic_stiffness()
```

I'm not satisfy by this user interface. It requires the users to order the operation in a way that can be hard to guess and to remember...

**Comment by ykshif** [comment]
Hi mancellin,
When I calculate the inertia of Rectangular, I find that the full mesh and clipped mesh are not the same.Then I calculate the inertia of Sphere,the result is the same.I don't know if it's a bug?
`rectangle1 = cpt.RectangularParallelepiped(size=(1.0, 1.0, 1.0), resolution=(10, 10, 10), center=(0, 0, 0),name="body1Name")`
`rectangle1.center_of_mass = (0,0,0)`
`rectangle1.add_all_rigid_body_dofs()`
`rectangle1.inertia_matrix = rectangle1.compute_rigid_body_inertia()`
![image](https://user-images.githubusercontent.com/111233162/187878178-9d52daee-c804-450f-afa4-9f3827ada900.png)
When I clipped mesh.
![image](https://user-images.githubusercontent.com/111233162/187878400-7fc5aa77-f62a-477c-8fae-e0faa566e159.png)

**Comment by sdillenburg** [comment]
I believe changing the procedure here does not yield a lot of additional value. Of course, the inertia matrix can be estimated from the geometry, but the actual data depends on the project specific weight distribution and is to be provided by the user. Maybe one could do something like `body.radii = [9.,26.5,27.8]` and then let Capytaine compile the full stiffness matrix based on the computed displacement:
```
M = np.eye(6)*self.disp_mass(rho=rho)
M[3:,3:] *= self.radii**2
```

**Comment by mancellin** [comment]
> When I calculate the inertia of Rectangular, I find that the full mesh and clipped mesh are not the same.Then I calculate the inertia of Sphere,the result is the same.I don't know if it's a bug?

If you are using v1.4, it might be #182.

**Comment by mancellin** [comment]
> I believe changing the procedure here does not yield a lot of additional value. Of course, the inertia matrix can be estimated from the geometry, but the actual data depends on the project specific weight distribution and is to be provided by the user.

Yes, for most real-life problems the inertia matrix will need to be provided by the user anyway.
But I believe it is important to have an exact computation of the rigid-homogeneous-body inertia for testing and teaching.

> Maybe one could do something like `body.radii = [9.,26.5,27.8]` and then let Capytaine compile the full stiffness matrix based on the computed displacement:
> 
> ```
> M = np.eye(6)*self.disp_mass(rho=rho)
> M[3:,3:] *= self.radii**2
> ```

It is possible to add a `compute_inertia_matrix_from_radii()` method to `FloatingBody`. If you'd like that, could you open a new issue with the motivation and examples (e.g. from the literature)?

**Comment by mancellin** [comment]
Mostly solved by #262.
